### DISCUSSION TITLE
Masking json path by applying predicate

### DISCUSSION THREAD

**Discussion by minubiyer** [logfellow/logstash-logback-encoder#1011] [Q&A]
I am using structured logging design to log data. When logging a particular data i would like to mask using predicate in json pointer.  For eg:
```java
log.info("Printing greeting detail object received in request",greeting);
```

```json
{"@timestamp":"2024-03-12 10:18:06,297","severity":"INFO","service":"Demo Application Service","class":"c.e.reactivewebservice.GreetingHandler","message":{"text":"Printing greeting detail object received in request ","details":{"Greeting":{"message":"Welcome to Phuket","name":"John","age":"26","world":{"name":null,"geoName":"phuk","geography":null,"longitude":230,"lat":400},"markets":[{"marketName":"spa","location":"mum","gstNo":"2345666"},{"marketName":"bugis","location":"sing","gstNo":"2245680087"},{"marketName":"flower","location":"washdc","gstNo":"112367886433"}]}}},"thread":"reactor-http-nio-6","pid":"29416","correlation-id":"","trace":""}
```

I am using logback.xml in which i am using MaskingJsonGeneratorDecorator, defining path
/message/details/Greeting/name ---- this will mask name field (irrespective of value)
/message/details/Greeting/age -- this will mask age field (similar to above)

I have requirement if field value contains 'null' then it should not mask (how to do this? As current class masks irrespective of value), is it possible to apply predicate /message/details/Greeting[@.name != 'null']/name --- Kindly let me know how to apply predicate or help me create custom masking component which i can apply to get the predicate support

**Comment by philsttr** [comment]
You'll need to use a [value masker](https://github.com/logfellow/logstash-logback-encoder?tab=readme-ov-file#identifying-field-values-to-mask-by-value) (as opposed to a [path masker](https://github.com/logfellow/logstash-logback-encoder?tab=readme-ov-file#identifying-field-values-to-mask-by-path)) if you want to mask based on field values.

**Comment by mgoyal0208** [comment]
Hi @minubiyer, I had a similar requirement, and I wrote a custom Field Masker class by implementing FieldMasker interface. However, this is causing an issue due to higher number to iteration in case the payload is huge. 

Please let me know if you have any other alternative solution for this. Thanks.
